### PR TITLE
Improved message when template from the cache is not available

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
fixes dotnet/templating#3124 
The message when the template content is not available is misleading

### Solution
Improved message, corrected command examples

### Checks:
- [x] Added unit tests
- [x] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style)

### CODE REVIEW

**Review by baronfel — APPROVED** [review]
This is a good change 👍 

As a note, I love the example-builder you have for generating command lines - it's very easy to use/read!

**Review by GangWang01 — APPROVED** [review]
It looks like using the error message on initiation guides user to resolve wrong template list from `dotnet new --list`.

File: src/Microsoft.TemplateEngine.Cli/TemplateInvoker.cs [hunk 1]
```diff
@@ -232,7 +234,29 @@ private async Task<NewCommandStatus> CreateTemplateAsync(TemplateCommandArgs tem
...
+                        {
+                            Reporter.Output.WriteLine(LocalizableStrings.TemplateCreator_Hint_Uninstall);
+                            Reporter.Output.WriteCommand(Example.For<UninstallCommand>(templateArgs.ParseResult).WithArgument(UninstallCommand.NameArgument, templatePackage.DisplayName));
+                            Reporter.Output.WriteLine();
+                            Reporter.Output.WriteLine(LocalizableStrings.TemplateCreator_Hint_Install);
+                            Reporter.Output.WriteCommand(Example.For<InstallCommand>(templateArgs.ParseResult).WithArgument(InstallCommand.NameArgument, templatePackage.DisplayName));
+                            Reporter.Output.WriteLine();
+                        }
+                    }
+                    catch
```
Comment (JanKrivanek) [inline src/Microsoft.TemplateEngine.Cli/TemplateInvoker.cs:256]: nit: the try-catch should be sufficient just for the first 2 lines.

**Review by JanKrivanek — APPROVED** [review]
Looks helpful!

**Pull request merged**